inc/fields.class.php

```diff
@@ -332,7 +331,7 @@ public static function updateVisibility($input) {
       }
 
       $questionToShow = [];
-      foreach ($input as $id => $value) {
+      foreach ($fields as $id => $value) {
```

Comment:
some field types don't set a $_POST value when they are empty (example: actors field). It is better to traverse the list of questions in the form rather than the $_POST data